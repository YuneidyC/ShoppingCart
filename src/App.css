@import 'tailwindcss';

@theme {
  --color-black: black;
  --shadow-products: 0px 0px 4px 4px rgb(0 0 0 / 0.05);
  --my-height-product: calc(100vh - 50px);
}
