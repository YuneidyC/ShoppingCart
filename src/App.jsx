import { useEffect, useState } from 'react';

import Navbar from './component/Navbar';
import ProductList from './component/ProductList';

import './App.css';

const initialState = {
    cart: [],
};

function App() {
    const serverUrl = 'https://fakestoreapi.com/products';
    const [products, setProducts] = useState([]);
    const [items, setItems] = useState(initialState);

    useEffect(() => {
        fetch(serverUrl)
            .then((response) => response.json())
            .then((data) => setProducts(data));
    }, []);

    const addToCart = (payload) => {
        console.log(payload);
        if (items.cart.length >= 0) {
            const found = items.cart.find((element) => {
                return element.id === payload.id;
            });
            if (found !== undefined) {
                found.qty++;
                setItems({
                    ...items,
                    cart: [...items.cart],
                });
            } else {
                payload.qty = 1;
                setItems({
                    ...items,
                    cart: [...items.cart, payload],
                });
            }
        }
    };

    const removeFromCart = (payload, indexValue) => {
        setItems({
            ...items,
            cart: items.cart.filter(
                (items, index) => items.id !== payload && index !== indexValue
            ),
        });
    };

    return (
        <>
                addToCart={addToCart}
                removeFromCart={removeFromCart}
                addToCart={addToCart}
        </>
    );
}

export default App;
