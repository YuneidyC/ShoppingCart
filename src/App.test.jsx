import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import App from './App';

describe('App component', () => {
    it('renders hello', () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot();
    });
});
