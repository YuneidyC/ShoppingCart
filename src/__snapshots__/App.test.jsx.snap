// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders hello 1`] = `
<div>
  <div
    class="flex flex-row bg-white px-[20px] h-[48px] w-full justify-between items-center shadow-md nav-bar"
  >
    <h1
      class="text-black text-[24px]"
    >
      Shopping Cart
    </h1>
    <button
      class="w-[30px] h-[30px] cursor-pointer"
    >
      <img
        alt="cart"
        class="w-[30px] h-[30px]"
        src="/src/assets/cart.png"
      />
    </button>
  </div>
  <ul
    class=" h-(--my-height-product) m-auto p-[50px] grid grid-cols-[repeat(auto-fit,_minmax(_200px,_1fr))] gap-x-3 gap-y-5"
  />
</div>
`;
